src/Axe/Managers/CompilerManager.cs

```diff
@@ -265,10 +276,105 @@ public CompilerManager(AxeDbContext context, INodeServices nodeServices) : base(
+                    // Generate the code for execution
+                    string code = model.SourceCode;
+                    if (codeBlock.TestCases[i].Input != null)
+                    {
+                        string[] inputArguments = codeBlock.TestCases[i].Input.Split(' ');
+                        code += "\nprint(main(";
+                        for (int j = 0; j < inputArguments.Length; j++)
+                        {
+                            code += inputArguments[j];
+                            if (j < inputArguments.Length - 1)
+                            {
+                                code += ", ";
+                            }
+                        }
+                        code += "));";
+                    }
+                    else
+                    {
+                        code += "\nprint(main());";
+                    }
+                    // Write the code into temporary file
+                    using (var pyFile = File.Create(pyFilePath))
+                    {
+                        using (var stream = new StreamWriter(pyFile))
+                        {
+                            stream.Write(code);
+                        }
+                    }
+                    // Create process for Python and execute script
+                    Process process = new Process();
+                    process.StartInfo = startInfo;
+                    process.Start();
+                    // Read data from terminal
+                    string error;
+                    using (var stream = process.StandardError)
+                    {
+                        error = await stream.ReadToEndAsync();
+                    }
+                    if (error.Length > 0)
+                    {
+                        return new Tuple<CodeBlockResult, string[]>(CodeBlockResult.Error, new string[] { error });
+                    }
+
+                    string output;
+                    using (var stream = process.StandardOutput)
+                    {
+                        output = await stream.ReadToEndAsync();
+                    }
+                    output = output.Replace("\r\n", ""); // Console returnes this symbol in the end of the string
+                    if (output != codeBlock.TestCases[i].Output)
+                    {
+                        failedTestCases.Add("Test case where input is " + (codeBlock.TestCases[i].Input ?? " ") + " and output is " + codeBlock.TestCases[i].Output + " was failed");
+                    }
+
+                    if (!process.HasExited)
+                    {
+                        process.Kill();
+                    }
+                }
+                if (File.Exists(pyFilePath))
+                {
+                    File.Delete(pyFilePath);
+                }
+
+                string[] info = new string[] { failedTestCases.Count + " of " + codeBlock.TestCases.Count + " testcases ended in failure" }.Union(failedTestCases).ToArray();
```

Comment:
why not insert string into `failedTestCases` at 0 position instead of making new array and then Union?
also I think  linq `Concat()` method is more appropriate here because Union removes duplicates